WooCommerce/Classes/ViewRelated/BottomSheet/ListSelector/BottomSheetListSelectorViewController.swift

```diff
@@ -78,6 +78,7 @@ UIViewController, UITableViewDataSource, UITableViewDelegate where Command.Model
             command.handleSelectedChange(selected: selected)
             tableView.reloadData()
         }
+        dismiss(animated: true, completion: nil)
```

Comment:
I looked at other places where this ViewController is used. It looks like the original expectation is to have the _presenting view controller_ dismiss the `BottomSheetListSelectorViewController`. Examples:

For the "More details" in a product:

https://github.com/woocommerce/woocommerce-ios/blob/c57b3472b0569f732a8e70728fcf42955cc746c5/WooCommerce/Classes/ViewRelated/Products/Edit%20Product/ProductFormViewController.swift#L565-L591

For the product type selection when adding a product: 

https://github.com/woocommerce/woocommerce-ios/blob/c57b3472b0569f732a8e70728fcf42955cc746c5/WooCommerce/Classes/ViewRelated/Products/Add%20Product/AddProductCoordinator.swift#L46-L53

Notice the `dismiss()` calls in those places. And there are a couple more that I have not listed.

Adding this dismissal here would negate those implementations. Or I don't know what else would happen if `dismiss()` is called twice. While we could remove those places where `dismiss()` is called, I think it's safer to follow the convention instead and manually call `dismiss()` where the sorting dialog is presented. 

To do that, it looks like we can change the implementation of [`ProductsSortOrderBottomSheetListSelectorCommand`](https://github.com/woocommerce/woocommerce-ios/blob/c57b3472b0569f732a8e70728fcf42955cc746c5/WooCommerce/Classes/ViewRelated/Products/BottomSheetListSelector/ProductsSortOrderBottomSheetListSelectorCommand.swift#L22) so it's more like [`ProductTypeBottomSheetListSelectorCommand`](https://github.com/woocommerce/woocommerce-ios/blob/c57b3472b0569f732a8e70728fcf42955cc746c5/WooCommerce/Classes/ViewRelated/Products/Edit%20Product/BottomSheetListSelector/ProductTypeBottomSheetListSelectorCommand.swift#L132). 

Notice that `ProductTypeBottomSheetListSelectorCommand` propagates the `onSelection` event during the `handleSelectedChange` event. That `onSelection` block dismisses the view. We can follow the same strategy with sorting. 

What do you think?

Reply:
That make a lot of sense, thanks for the pointers and the detailed walkthrough!